ql/src/java/org/apache/hadoop/hive/ql/exec/vector/VectorDeserializeRow.java

```diff
@@ -933,12 +1207,20 @@ private void storeUnionRowColumn(ColumnVector colVector,
     unionColVector.isNull[batchIndex] = false;
     unionColVector.tags[batchIndex] = tag;
 
-    storeComplexFieldRowColumn(
+    deserializer.storeComplexFieldRowColumn(
         colVectorFields[tag],
         unionHelper.getFields()[tag],
         batchIndex,
         canRetainByteRef);
-    deserializeRead.finishComplexVariableFieldsType();
+    deserializer.deserializeRead.finishComplexVariableFieldsType();
+  }
+
+  abstract static class VectorBatchDeserializer {
+    abstract void store(ColumnVector colVector, Field field, int batchIndex, boolean canRetainByteRef,
+                            VectorDeserializeRow deserializer) throws IOException;
```

Comment:
Why VectorDeserializerRow needs be passed here again? ("this" references in other places as well). If you remove "static" class declaration in VectorBatchDeserializer children, you may not need to pass this.  And the patch would become lot lesser changes?